common/log.c

```diff
@@ -530,17 +530,37 @@ internal_log_is_enabled_for_level(const enum logLevels log_level,
     {
         return 0;
     }
-
-    /* Is there at least one log destination which will accept the message based on the log level? */
-    return (g_staticLogConfig->fd >= 0
-               && ((override_destination_level && log_level <= override_log_level)
-                  || (!override_destination_level && log_level <= g_staticLogConfig->log_level)))
-           || (g_staticLogConfig->enable_syslog
-               && ((override_destination_level && log_level <= override_log_level)
-                  || (!override_destination_level && log_level <= g_staticLogConfig->syslog_level)))
-           || (g_staticLogConfig->enable_console
-               && ((override_destination_level && log_level <= override_log_level)
-                  || (!override_destination_level && log_level <= g_staticLogConfig->console_level)));
+    else if (g_staticLogConfig->fd < 0
+            && !g_staticLogConfig->enable_syslog
+            && !g_staticLogConfig->enable_console)
+    {
+        /* all logging outputs are disabled */
+        return 0;
+    }
+    else if (override_destination_level)
```

Comment:
I'm happy with either multiple returns in a function, or if / else if with a single return at the end. Here you've mixed styles somewhat. It'll work fine, but I think there's a chance that in the future a static analysis will moan about an unreachable 'else' branch following a return. 

I'd suggest either removing the unnecessary `else` statements, or using a single return - whatever seems more natural to you.

Reply:
@matt335672 I've implemented this feedback using multiple returns by only have a single level of nesting for the if/else. Please let me know if I correctly interpreted your feedback, or if you'd prefer that I use the single return at the end style.

Reply:
Sorry - I've probably not been very clear.

My point was that in this kind of construction:-

<pre>
if (&lt;something&gt;)
{
    return 1;
}
<b>else</b> if (&lt;something else&gt;)
...
</pre>

the `else` in bold is superfluous and not required. At some stage I'd expect a static analysis tool to say something about it. Does that make more sense?

The nesting style you use, or whether you use a single return or multiple returns is completely up to you. I've got my own preferences but they're not relevant at all.